controller/raft/raft.go

```diff
@@ -119,6 +122,10 @@ func (self *Controller) Dispatch(cmd command.Command) error {
 		}
 	}
 
+	if self.Mesh.IsReadOnly() {
```

Comment:
Can you add some text about why it's read-only? (either 'upgrade in progress' or 'different versions detected in cluster'). Leaning towards different versions, since that's most explicit.

Don't need Errorf because you're not injecting any params. Can just use error.New
Also general convention is lowercase for errors.